tasks/code_sync_status.rb

```diff
@@ -0,0 +1,93 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+
+# Puppet Task Name: code_sync_status
+require 'net/https'
+require 'uri'
+require 'json'
+require 'puppet'
+
+# CodeSyncStatus task class
+class CodeSyncStatus
+  def initialize(params)
+    @params = params
+  end
+
+  def execute!
+    puts syncstatus.to_json
+  end
+
+  def https
+    https = Net::HTTP.new('localhost', '8140')
+    https.use_ssl = true
+    https.cert = @cert ||= OpenSSL::X509::Certificate.new(File.read(Puppet.settings[:hostcert]))
+    https.key = @key ||= OpenSSL::PKey::RSA.new(File.read(Puppet.settings[:hostprivkey]))
+    https.verify_mode = OpenSSL::SSL::VERIFY_NONE
+    https
+  end
+
+  def apistatus
+    status = https
+    # Only debug level includes code sync details
+    status_request = Net::HTTP::Get.new('/status/v1/services?level=debug')
+    JSON.parse(status.request(status_request).body)
+  end
+
+  def checkenvironmentlist(environments, requestedenvironments)
+    environmentstocheck = []
+    # If all was passed as an argument we check all visible environments
+    if requestedenvironments.any? { |s| s.casecmp('all') == 0 }
+      environmentstocheck = environments
+    # Else check each requested environment to confirm its a visible environment
+    else
+      requestedenvironments.each do |environment|
+        environments.any? { |s| s.casecmp(environment.to_s) == 0 } || raise("Environment #{environment} is not visible and will not be checked")
+        environmentstocheck << environment
+      end
+    end
+    environmentstocheck
+  end
+
+  def checkenvironmentcode(environment, servers, statuscall)
```

Comment:
What about this method name?

```suggestion
  def check_environment_code(environment, servers, status_call)
```

Reply:
done